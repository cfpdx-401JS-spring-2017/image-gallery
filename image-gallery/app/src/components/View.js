/*import React, { Component } from 'react';
import Intro from './IntroView'
import PropTypes from 'prop-types';
import AddImage from './AddImage'



export default class View extends Component {

    constructor(props) {
        super(props);

        this.state = {
            CurrentView: Intro
        }

    }
    //use static if defining proptypes within the class.  
    // if function  statement looks like: View.propTypes.string.isrequired
    static propTypes = {
        images: PropTypes.array.isRequired,
        onDelete: PropTypes.func.isRequired,
        handleAdd: PropTypes.func.isRequired
    }

    changeView(View = Intro) {
        this.setState({ CurrentView: View });
    }


    render() {

        const { CurrentView } = this.state;
        return (
            <div>

                <CurrentView
                    images={this.props.images}
                    onDelete={this.props.onDelete}
                    handleAdd={this.props.handleAdd} />

                {CurrentView !== Intro &&  
                <div className="control">
                    <AddImage handleAdd={this.props.handleAdd}
                    />
                </div>}

            </div>

        )
    }

}

//Q: do I need <AddImage in this component?*/