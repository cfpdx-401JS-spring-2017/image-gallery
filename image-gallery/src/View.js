import React, { Component } from 'react';
import List from './galleries/List';
import Single from './galleries/Single';
import Thumbnail from './galleries/Thumbnail';

export default class Veiw extends Component {

    constructor(props) {
        super(props);

        //Q: Why do we capitalize Current Scene? BC it will have a value of an imported component (List, THUMB, SINGLE).

        this.state = {
            CurrentView: null
        }

    }

    changeView( View = null) {
        this.setState({ CurrentView: View});
    }


    render() {

        const { CurrentView } = this.state;
        return (
            <div>
                <header>
                    <h1>WELCOME TO THE FUZZY BUNNY IMAGE GALLERY</h1>
                </header>
                <h2>Check out all the bunnies...</h2>
                <span>
                    <button 
                    onClick={() => this.changeView(List)}
                    >CHECK OUT LIST VIEW</button>
                </span>
                <span>
                    <button 
                    onClick={() => this.changeView(Thumbnail)}
                    >CHECK OUT THUMBNAIL VIEW</button>
                </span>
                <span>
                    <button 
                    onClick={() => this.changeView(Single)}
                    >CHECK OUT SINGLE VIEW</button>
                </span>
                {CurrentView && <CurrentView 
                    images={this.props.images} />
                }
        </div>
             
    )
}

}