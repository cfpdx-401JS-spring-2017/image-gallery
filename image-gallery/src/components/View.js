import React, { Component } from 'react';
import List from './List';
import Gallery from './Gallery';
import Thumbnail from './Thumbnail';
import Intro from './IntroView'
import PropTypes from 'prop-types';


export default class Veiw extends Component {

    constructor(props) {
        super(props);

        this.state = {
            CurrentView: Intro
        }

    }
    //use static if defining proptypes within the class.  
    // if function  statement looks like: View.propTypes.string.isrequired
    static propTypes = {
        images: PropTypes.array.isRequired
    }

    changeView(View = Intro) {
        this.setState({ CurrentView: View });
    }


    render() {

        const { CurrentView } = this.state;
        return (
            <div className="header-container">
                <header className="header-title"
                    onClick={() => this.changeView(Intro)}>
                    <h1 > The Zephr Originals. </h1>
                    <h2>Z-Girls & Z-Boys of the 70's</h2>
                </header>

                <span>
                    <button className="view-buttons"
                        onClick={() => this.changeView(List)}
                    >CHECK OUT LIST VIEW</button>
                </span>
                <span>
                    <button className="view-buttons"
                        onClick={() => this.changeView(Thumbnail)}
                    >CHECK OUT THUMBNAIL VIEW</button>
                </span>
                <span>
                    <button className="view-buttons"
                        onClick={() => this.changeView(Gallery)}>CHECK OUT GALLERY VIEW</button>
                </span>


                {CurrentView && <CurrentView
                    images={this.props.images} />
                }

            </div>

        )
    }

}