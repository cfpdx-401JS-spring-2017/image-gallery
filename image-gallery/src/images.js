import lordsOfDogtown from './photos/lordsOfDogtown.jpg';
import jayAdams from './photos/jayAdams.png';
import peggyOki from './photos/peggyOki.jpg';
import mikeKolar from './photos/mikeKolar.jpg';
import peggyTurner from './photos/peggyTurner.jpg';
import lauraThornhill from './photos/lauraThornhill2.jpg';
import rampage from './photos/rampage.jpg';
import shortId from 'shortid';
// let shortId = require('shortid');

const images = [
    {
        title: 'Z Boys',
        description: 'Zephyr skate team 1970\'s',
        url: lordsOfDogtown,
        _id: shortId.generate()
    },

    {
        title: 'Turn Only',
        description: 'Jay Adams gets low',
        url: jayAdams,
        _id: shortId.generate()
    },

    {
        title: 'Peggy Does Venice',
        description: 'Peggy Oki repping Zephr',
        url: peggyOki,
        _id: shortId.generate()
    },

    {
        title: 'Jumping Homies',
        description: 'Mike Kolar 1970\'s',
        url: mikeKolar,
        _id: shortId.generate()
    },

    {
        title: "Tippin' Toes",
        description: 'Laura Thornhill in skate competition 1970\'s',
        url: lauraThornhill,
        _id: shortId.generate()
    },

    {
        title: "Van Fans",
        description: 'Peggy Turner repping Vans in skate competition',
        url: peggyTurner,
        _id: shortId.generate()
    },

    {
        title: "Jumping Homies 2.0",
        description: 'Rampage Expo 1970\'s',
        url: rampage,
        _id: shortId.generate()
    }

];

export default {
    get() { return Promise.resolve(images) }
}