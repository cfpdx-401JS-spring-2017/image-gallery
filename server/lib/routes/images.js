const Router = require('express').Router;
const router = Router();
const Image = require('../models/image');

router

.get('/api/images',(res, req, next) => {
  Image.find()
    .then(images => res.send(images))
    .catch(next);
})

.get('/id', (req, res, next) => {
  const id = req.params.id;
  Image.findById(id)
    .then(image => res.send(image))
    .catch(next);
})

.post('/', (req, res, next) => {
  new Image(req.body)
    .save()
    .then(image => res.send(image))
    .catch(next);
})

.put('/id', (req, res, next) => {
  Image.findByIdAndUpdate(req.params.id, req.body, {new: true})
    .then(image => res.send(image))
    .catch(next);
})

.delete('./id', (req, res, next) => {
  Image.findByIdAndRemove(req.params.id)
    .then(response => {
      res.send({removed: response ? true : false });
    })
    .catch(next);
});

module.exports = router;
