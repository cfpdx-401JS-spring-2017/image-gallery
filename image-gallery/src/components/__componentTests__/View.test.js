import React from 'react';
import ReactDOM from 'react-dom';
import View from '../View';
import renderer from 'react-test-renderer';
import images from '../../imagesData';

//Q: how do I pass onDelete and HandleAdd here?




it('renders View view without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<View  
  images={images}/>, div);
});

it('renders a snapshot', () => {
  const tree = renderer.create(<View images={images}/>).toJSON();
  expect(tree).toMatchSnapshot();
});