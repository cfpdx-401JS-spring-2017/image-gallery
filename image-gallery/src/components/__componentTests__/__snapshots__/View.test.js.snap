// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="header-container"
>
  <header
    className="header-title"
    onClick={[Function]}
  >
    <h1>
       The Zephr Originals. 
    </h1>
    <h2>
      Z-Girls & Z-Boys of the 70's
    </h2>
  </header>
  <span>
    <button
      className="view-buttons"
      onClick={[Function]}
    >
      CHECK OUT LIST VIEW
    </button>
  </span>
  <span>
    <button
      className="view-buttons"
      onClick={[Function]}
    >
      CHECK OUT THUMBNAIL VIEW
    </button>
  </span>
  <span>
    <button
      className="view-buttons"
      onClick={[Function]}
    >
      CHECK OUT GALLERY VIEW
    </button>
  </span>
  <div
    className="intro-image brighten"
  >
    <a
      className="video"
      href="https://www.youtube.com/watch?v=cKC-XBwQ_pA"
    >
      <img
        alt="standing on boards"
        src="introImage.jpg"
      />
    </a>
    <span
      className="play-video"
    >
      <h1>
        PLAY VIDEO
      </h1>
    </span>
  </div>
</div>
`;
